src/app/dataset/depublication/depublication.component.spec.ts

```diff
@@ -117,15 +118,56 @@ describe('DepublicationComponent', () => {
     });
 
     it('should submit the text', () => {
+      const datasetId = '123';
+      component.datasetId = datasetId;
       component.dialogInputOpen = true;
-      component.datasetId = '123';
+
       component.onSubmitRawText();
       expect(component.dialogInputOpen).toBeTruthy();
-      component.formRawText.patchValue({ recordIds: 'http://record/id' });
+      component.formRawText.patchValue({ recordIds: `http://${datasetId}/${recordId}` });
       component.onSubmitRawText();
       expect(component.dialogInputOpen).toBeFalsy();
     });
 
+    it('should validate the record ids', () => {
+      const datasetId = '123';
+      component.datasetId = datasetId;
+
+      expect(component.validateRecordIds(frmCtrl(recordId))).toBeFalsy();
+      expect(component.validateRecordIds(frmCtrl(`/${recordId}`))).toBeFalsy();
+      expect(component.validateRecordIds(frmCtrl(`${datasetId}/${recordId}`))).toBeFalsy();
+      expect(component.validateRecordIds(frmCtrl(`${recordId}/${datasetId}`))).toBeTruthy();
+      expect(component.validateRecordIds(frmCtrl(`${datasetId}/${recordId}`))).toBeFalsy();
+      expect(component.validateRecordIds(frmCtrl(`/${datasetId}/${recordId}`))).toBeFalsy();
+      expect(component.validateRecordIds(frmCtrl(`//${datasetId}/${recordId}`))).toBeTruthy();
+      expect(component.validateRecordIds(frmCtrl(`http://${datasetId}/${recordId}`))).toBeFalsy();
+      expect(
+        component.validateRecordIds(frmCtrl(`http://path/${datasetId}/${recordId}`))
+      ).toBeFalsy();
+      expect(
+        component.validateRecordIds(frmCtrl(`https://path/${datasetId}/${recordId}`))
+      ).toBeFalsy();
+      expect(
+        component.validateRecordIds(frmCtrl(`https://path/${datasetId}/${recordId}/`))
+      ).toBeTruthy();
+      expect(
+        component.validateRecordIds(
+          frmCtrl(`
+        https://path/${datasetId}/${recordId}
+        https://path/${datasetId}/${recordId}
+      `)
+        )
+      ).toBeFalsy();
+      expect(
+        component.validateRecordIds(
+          frmCtrl(`
+        https://path/${datasetId}/${recordId}/
+        https://path/${datasetId}/${recordId}
+      `)
+        )
+      ).toBeTruthy();
+    });
```

Comment:
I can come up with some others. How about checking the following:
- `http://www.server.com//path1/path2/${datasetId}/${recordId}` (should pass)
- `/${datasetId}/not_the_dataset/${recordId}` (should fail)
- values with spaces in the url (should fail)